### ISSUE TITLE
Artifacts when using some parameters on x265

### ISSUE THREAD

**Issue by p4thfind3r25** [staxrip/staxrip#1393]
**Describe the bug**
Using StaxRip"x265" for an 2160p.UHD.HDR Encode with "Recursion Skip"-Mod 2(RD LEVEL 0-6 CU edge density),no matter of other Settings used, the Encoded Video Stream has at least one section with heavy compression artifacts(I believe is "Blocking") which has a length of aprox. 10 sec.(Picture showing type of defect is attached)
I tried to use "rskip Mod 2" to encode many files like the one in the attached examples resulting same artifacts present in the Encoded Video stream. 

**Expected behavior**
I expect a clean Encoded Video Stream(no artifacts).

**How to reproduce the issue**
Load SOURCE FILE in StaxRip >>Crop as necessary and :
1. Go to 'Options >>'Analysis'
2. Click on 'Recursion Skip' --> 'Mod 2'
3. Using 'x265' encoder.
4. 'AviSynth  being used.
5. No Filter(s) x is being used.

**Provide information**
Encoding process ends successfully-No error message.
I discovered accidentally the moment when the full encoded file has artifacts and made a cut (+5min, -5min)from that moment and used as a test file reproducing the defect every time no matter variation in Settings. I consider this matter important because huge reduction in the encoding time with a minimal loss in quality(when using rskip-Mod 2).
I want to tell you that other encoders using same Settings but not using StaxRip are getting clean Video streams.(They are using probably "x265" as an stand alone software with command lines).I use StaxRip because I,m not good using "Avisinth" and "x265" through CLI.
- Used StaxRip version : 2.38.7 and 2.40.0

Attached Files:

![TYPE OF GLITCH ON TEST - RESIZED IMAGE - 1 and 2 and 3 and 4](https://github.com/staxrip/staxrip/assets/172151380/a3039d75-1beb-4b06-bc1d-0578bab60bce)

**Comment by Dendraspis** [comment]
Before I answer on your issue:
Can you tell me which app you have used to verify those artifacts?

**Comment by p4thfind3r25** [comment]
dvdfab player

**Comment by Dendraspis** [comment]
Do me a favor and open the resulting video with artifacts with StaxRip, just like you did with the original source.
Then open the `Preview` and jump to the scenes in question.

Can you verify, that there are no issues that way?

**Comment by p4thfind3r25** [comment]
I loaded one of the tests(TEST 3) into StaxRip ,went in "Preview" mode and played the affected section using StaxRip embeded player(mpc-hc) at normal speed and also a frame by frame image advance also on the affected section. In both instances the defect was no more present.

**Comment by Dendraspis** [comment]
Okay, so:

StaxRip is just a GUI for frame servers and encoders, meaning that every issue coming from one of those tools, which is clearly the case here, is related to that particular tool and has nothing to do with StaxRip. That said this is not the right place to report such an issue with `x265`.

Because you have posted a screenshot, I recognized this issue that was discussed recently on Discord about different parameters. It turns out, that it should be an issue with the filters, that are used by some players to read the video file. In case of `MPC-HC` you can solve this issue by going to `Options » Internal filter » Video-Decoder » Hardware Acceleration » Hardware Decoder` and set it to `DXVA (native)`, then those issues should be gone.


> StaxRip embeded player(mpc-hc) 

I guess you meant `mpv.net`, which was also confirmed to have no playback issues.

**Comment by p4thfind3r25** [comment]
For mpc-hc: video decoder is already set to dxva(native) and the defect persists. The only way to obtain a clean playable file is to set rskip to mod 1 during encoding process. With rskip set to mod 2 the issue is present no matter how other settings are modified. So the issue is rskip mode 2. 
Interesting is that I transferred the encoded files with defects on an external HDD and played them connecting the HDD directly to TV set: the defect disappeared. Also I connected HDD to an external Video player(ZIDOO) connected to TV and played the files: the defect disappeared. Playing the files through PC (no matter the player) always defect is present. All this mess is valid only for files encoded with rskip mod 2. Must be a bug or a setting which I miss or is the wrong value in my settings.

**Comment by Dendraspis** [comment]
Like I said, this is a player/reader issue, otherwise your TV would also show artifacts.

Interestingly you're the first person telling that `DXVA (native)` doesn't work. Also strange that...

> video decoder is already set to dxva(native)

...because the default is `DXVA(copy)`.

**Comment by apophis906** [comment]
> For mpc-hc: video decoder is already set to dxva(native) and the defect persists. The only way to obtain a clean playable file is to set rskip to mod 1 during encoding process. With rskip set to mod 2 the issue is present no matter how other settings are modified. So the issue is rskip mode 2. Interesting is that I transferred the encoded files with defects on an external HDD and played them connecting the HDD directly to TV set: the defect disappeared. Also I connected HDD to an external Video player(ZIDOO) connected to TV and played the files: the defect disappeared. Playing the files through PC (no matter the player) always defect is present. All this mess is valid only for files encoded with rskip mod 2. Must be a bug or a setting which I miss or is the wrong value in my settings.

I just saw this and thought I might offer some advice. I have a feeling you are using an nvidia card for playback. If that is the case then changing to software decoding will solve the problem. It is not Staxrip or x265, it is the nvidia decoder chip used in every card except the 950. They all show glitches with an HEVC video track in random places for no reason with various settings. This was even brought up in the lav github [https://github.com/Nevcairiel/LAVFilters/issues/534](url). This was also the reason that The Beastmaster was reported to have a playback glitch on stand alone players. It seems some brands use the same type of chip that nvidia uses. One person reported they had the glitch with an nvidia card and when they changed to an amd card the glitch went away. I set my playback to software because I got tired of thinking that I was encoding things with wrong settings or something. When in fact it is just the video card that does not like certain HEVC settings in certain situations. The reason it is 10 seconds is because it needed another refresh frame to flush the buffer of the error and start a new frame. So all in all your settings are not the problem, the nvidia card is. The fact that it plays fine on your TV and external player are proof. So your best bet is to do what I have done and just use software decoding for all HEVC and you wont see any glitches.
